### ISSUE TITLE
Override API endpoint URL

### ISSUE THREAD

**Issue by ianmaddox** [google/google-api-nodejs-client#621]
With preventExtensions turned on, it is not possible to duck-punch the ability to override API endpoint URLs. We need to do this because we are proxying GA requests so they can be properly rate limited and accounted for. Ideally, one would be able to add a key called 'url' to the params object passed into apirequest.createAPIRequest(). If provided, it can override the existing value.

**Comment by jmdobry** [comment]
@enobufs I'm thinking #623 ought to be able to solve this for you (allow you to pass `url: 'my_custom_url.com'`).

**Comment by jmdobry** [comment]
I released 12.4.0, give it a try:

```js
urlshortener.url.get({
  shortUrl: 'http://asdf4231'
}, {
  url: 'https://myproxy.com/urlshortener/v1/url'
}, function (err, response) {
  // ...
});
```
